setup.cfg

```diff
@@ -64,7 +64,7 @@ warn_incomplete_stub = true
+[mypy-homeassistant.bootstrap,homeassistant.components,homeassistant.config_entries,homeassistant.config,homeassistant.const,homeassistant.core,homeassistant.data_entry_flow,homeassistant.exceptions,homeassistant.loader,homeassistant.__main__,homeassistant.monkey_patch,homeassistant.requirements,homeassistant.setup,homeassistant.util,homeassistant.auth.*,homeassistant.components.automation.*,homeassistant.components.binary_sensor.*,homeassistant.components.calendar.*,homeassistant.components.cover.*,homeassistant.components.device_automation.*,homeassistant.components.frontend.*,homeassistant.components.geo_location.*,homeassistant.components.group.*,homeassistant.components.history.*,homeassistant.components.http.*,homeassistant.components.image_processing.*,homeassistant.components.integration.*,homeassistant.components.light.*,homeassistant.components.lock.*,homeassistant.components.mailbox.*,homeassistant.components.media_player.*,homeassistant.components.notify.*,homeassistant.components.persistent_notification.*,homeassistant.components.proximity.*,homeassistant.components.remote.*,homeassistant.components.scene.*,homeassistant.components.sensor.*,homeassistant.components.sun.*,homeassistant.components.switch.*,homeassistant.components.systemmonitor.*,homeassistant.components.tts.*,homeassistant.components.vacuum.*,homeassistant.components.water_heater.*,homeassistant.components.weather.*,homeassistant.components.websocket_api.*,homeassistant.components.zone.*,homeassistant.helpers.*,homeassistant.scripts.*,homeassistant.util.*,script.*]
```

Comment:
I don't think that we should type check our scripts.

Reply:
I (seriously) think that we should eventually type check everything :) And because the scripts already pass, and they do contain type hints, I think there's no harm in just starting to do it.

But there are three levels on which to do this:
- with the strict ruleset we have in place (that giant setup.cfg section name above)
- with the ignore errors fallback ruleset we have in place
- not at all

I think at least the ignore errors level should be run for everything. It'll just error out on fatal ones, parser etc level. That's good to have for everything. This is essentially what would happen if I leave the rest of this PR as is, but just remove script.* from the above mentioned line.

On the other hand, things we _really_ _really_ don't want mypy to look at (the not at all category), we need to add them to pre-commit's exclude.

Reply:
Anyway, removed script.* from the above for now.

BTW it would be better to use pre-commit in CI to check stuff too, to reduce number of different configs and bring dev setups closer to CI. Will have a look at this sometime.

Reply:
I remember @frenck having a repo somewhere where he was using pre-commit to do all the checks. I thought that was pretty smart. I agree that we should keep them aligned. 

The reason why I don't us to typecheck template files, is that they are templates. They might not be full pages, things are being changed on them using search/replace. I wouldn't want typecheck to get in our way of progressing how our scaffold scripts are used.

Reply:
Yes, I have set up pre-commit for CI elsewhere, works nice.

I've removed running mypy on scripts now, but the pre-commit config exclusion is still needed so that it doesn't get run on the templates when they change.